engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/MatchPredicateFRMatchingStep.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.predicate.dsl;
+
+import org.hibernate.search.engine.search.common.ValueConvert;
+
+/**
+ * The step in a "match" predicate definition where the value to match can be set.
+ *
+ * @param <N> The type of the next step.
+ */
+public interface MatchPredicateFRMatchingStep<T, N extends MatchPredicateOptionsStep<?>> {
```

Comment:
Obviously you could simply add generic type parameters to existing DSL steps, but just as obviously this would break compatibility with existing apps, which technically we're not allowed to do.

I wish I had put things like [this](https://github.com/hibernate/hibernate-search/blob/91ed4895382d2db8c78ba4fb8afda9ff1d10f9fc/engine/src/main/java/org/hibernate/search/engine/search/projection/dsl/ExtendedSearchProjectionFactory.java#L13-L14) in every DSL step, but now it's too late for that :/

So, additional interfaces are the way to go... but. IMO we'd need to plan for merging the new interfaces with the existing ones eventually. We can't keep duplicating them forever.

I would suggest making the new interfaces superinterfaces of the existing ones.
I.e. `MatchPredicateMatchingStep<N> extends MatchPredicateMatchingGenericStep<T, N, Object>`. That way:

1. `MatchPredicateMatchingStep` does not need to define any method. You're not duplicating that code and javadoc.
2. You can deprecate `MatchPredicateMatchingStep`, suggesting to use `MatchPredicateMatchingGenericStep` instead, and in the next major you can remove `MatchPredicateMatchingStep` altogether.